pkg/sql/parser/sql.y

```diff
@@ -6563,6 +6527,28 @@ signed_iconst:
     $$.val = &tree.NumVal{Value: constant.UnaryOp(token.SUB, $2.numVal().Value, 0)}
   }
 
+// signed_iconst64 is a variant of signed_iconst which only accepts (signed) integer literals that fit in an int64.
+signed_iconst64:
```

Comment:
i didn't totally follow the discussion around this one but is it intentional that this rule isn't used?

Reply:
Added comment with more explanation. It's used in a later commit than the one that introduces it.